### PR TITLE
feat: intergrate logging function

### PR DESCRIPTION
**Pull request by hanzili**
## Related Issue
 When executing installation script, user can easily miss the log message and continue using keploy with a fail initlization (that's what happened to me T^T

Closes: #1701 

#### Describe the changes you've made
- Introduced `log_message` for standardized and color-coded logging.
- Enhanced error handling to pause script execution and ensure user have acknowledged the errors.

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Code style update (formatting, local variables)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Please let us know if any test cases are added

NA

#### Describe if there is any unusual behaviour of your code(Write `NA` if there isn't)

NA

## Checklist:

- [x] My code follows the style guidelines of this project.
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have made corresponding changes to the documentation.
- [x] My changes generate no new warnings.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [x] New and existing unit tests pass locally with my changes.

## Screenshots (if any)
Now, when there is an error level logging, the execution will be paused until user acknowledge the message.
image

### CODE REVIEW

**Comment by hanzili** [comment]
I have read the CLA Document and I hereby sign the CLA

**Comment by shivamsouravjha** [comment]
@hanzili it would be better if Keploy logs that docker daemon isn't running, reason being if someone has keploy installed, restarted laptop or stopped docker manually, once they try to run keploy they would face similar issue.

So it would be better approach to rather log in keploy

**Comment by hanzili** [comment]
> @hanzili it would be better if Keploy logs that docker daemon isn't running, reason being if someone has keploy installed, restarted laptop or stopped docker manually, once they try to run keploy they would face similar issue.
> 
> So it would be better approach to rather log in keploy

Sorry for getting back so late! Do you mean checking is docker is running when keploy starts?  Could you explain more about the issue when someone restarted laptop or stopped docker manually as you mentioned? It sounds like a different issue than the one I encountered. My issue is that the keploy installation script does not initialize everything needed to run keploy (like creating `keploy-network`) and I did not realize that when using keploy.

File: keploy.sh [hunk 1]
```diff
@@ -0,0 +1,241 @@
...
+            # Add sudo to docker
+            network_alias="sudo"
+        fi
+        if ! $network_alias which docker &> /dev/null; then
+            log_message "ERROR" "Docker not found on device, please install docker to use Keploy."
+            return 0
+        fi
+        # Check if docker is running
+        if ! $network_alias docker info &> /dev/null; then
+            log_message Keploy only supports intercepting and replaying docker containers on macOS, and requires Docker to be installed and running. Please start Docker and try again.
```
Comment (lorenzocastelli) [inline keploy.sh:146]: @hanzili  Missing Log level ;)

**Comment by shivamsouravjha** [comment]
closing the PR as issue is solved

**Pull request closed without merging**